var/spack/repos/builtin/packages/py-shortbred/package.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class PyShortbred(PythonPackage):
+    """ShortBRED is a system for profiling protein families of interest at
+    very high specificity in shotgun meta'omic sequencing data."""
+
+    homepage = "https://huttenhower.sph.harvard.edu/shortbred/"
+    pypi = "shortbred/shortbred-0.9.5.tar.gz"
+
+    version("0.9.5", sha256="a6ac09b858f14e2c0b8622b122ec91e5d02d32c12429cad66626d7ef26df10d5")
+
+    depends_on("python@3:", type=("build", "run"))
```

Comment:
Where do you see that Python 3 is required? According to the README, it's still 2.7.9

Reply:
Sigh, so this software is kind of a mess and basically abandoned. They don't advertise the existence of 0.9.5 other than a github tag and uploading it to PyPi (which they don't mention on their website or README).

If you look through the commit history, they basically just changed print statements to make it python 3 compatible, since the previous package spit out errors if you used python 3 with it. I'm not sure if they made other changes, so I just fixed it there in case (and the PyPi page mentions 3.4, but it's not in the setup.py as a requirement).

There are also some other problems in the software like dependency check errors that are the result of typos, so I don't even know how necessary all the version requirements are.

Reply:
`print(...)` should work for both Python 2.7 and 3, so if there's no other reason to change it, I would leave it as `python@2.7.9:` (which includes 3)

Reply:
Are you using the new concretizer? If it doesn't work with the new concretizer, I'm guessing it isn't possible to build with Python 2 anyway.

Reply:
I believe so:

```
* **Spack:** 0.19.0.dev0 (75dd122175c48bf1c378191f79f88e08b02890ee)
* **Python:** 3.6.8
* **Platform:** linux-rocky8-cascadelake
* **Concretizer:** clingo
```

Reply:
The problem is that `py-biopython` depends on `python@3.6:`, so it's impossible to test Python 2 anyway. My suggestion would be to leave the dep as is (2.7.9+)

Reply:
oh, I think that's a mistake with the `py-biopython` package then. v1.65 was released in 2014 and supports python 2 and python 3:

https://github.com/biopython/biopython/tree/biopython-165